WalletWasabi.Tests/Helpers/ArenaBuilder.cs

```diff
@@ -0,0 +1,70 @@
+using Moq;
+using NBitcoin;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using WalletWasabi.BitcoinCore.Rpc;
+using WalletWasabi.WabiSabi.Backend;
+using WalletWasabi.WabiSabi.Backend.Banning;
+using WalletWasabi.WabiSabi.Backend.Rounds;
+
+namespace WalletWasabi.Tests.Helpers
+{
+	/// <summary>
+	/// Builder class for <see cref="Arena"/>.
+	/// </summary>
+	public class ArenaBuilder
+	{
+		public TimeSpan? Period { get; set; }
+		public Network? Network { get; set; }
+		public WabiSabiConfig? Config { get; set; }
+		public IRPCClient? Rpc { get; set; }
+		public Prison? Prison { get; set; }
+
+		/// <param name="rounds">Rounds to initialize <see cref="Arena"/> with.</param>
+		public Arena Create(params Round[] rounds)
```

Comment:
The suggested code is not actually allowed as it would result in a compile-time error. 

An alternative is to have:

```suggestion
		public Arena Create(IEnumerable<Round> rounds) // without "params"
```

However, `params` are quite useful in tests as they allow you to write `.Create(myRound)`, `.Create(round1, round2)`. etc.

> Apart from this type suggestion, arguably creating `Round` in the unit test is also a violation of the API, since the round object represents Arena internal mutable state, and so should be created by the `Arena` object, not created independently and then injected, as unfortunately is quite pervasive in the unit tests due to the way `WabiSabiFactory` works.

I agree. My approach here is to get concept ACKs first for making `Arena.Prison` property private. If agreed, I can continue with other public properties. That would be, let's say, step one. Step two would be to tackle "Arena.Rounds" not to be `public` but I don't have any clear idea how to do that at the moment (certainly glad to get feedback on that).

Also I don't think that this PR makes it significantly harder to tackle `Arena.Rounds` as I don't really change the API that significantly (IMO).

Reply:
Ah nevermind, my rationale for enumerable doesn't really make sense with `params`, just ignore it